.agents/skills/code-flow/scripts/code_flow.py

```
@@ -0,0 +1,297 @@
+    for source in captured['sources']:
+        path = safe_path(root, source['path'])
+        require(path.is_file(), f'source file missing: {source["path"]}')
+        require(path.stat().st_size <= 2_000_000, 'source file exceeds 2 MB')
+        if source['path'] not in files:
+            raw = path.read_bytes()
+            files[source['path']] = (digest(raw), raw.decode('utf-8').splitlines())
+        sha, lines = files[source['path']]
+        require(source['end'] <= len(lines), f'source range outside file: {source["id"]}')
+        source['excerpt'] = '\n'.join(lines[source['start'] - 1:source['end']])
+        source['sha256'] = sha
+    result = subprocess.run(['git', '-C', str(root), 'rev-parse', 'HEAD'], capture_output=True, text=True)
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)